File: Graph/1012.py
import sys
d = [(1,0),(-1,0),(0,1),(0,-1)]
sys.setrecursionlimit(10**6)
def sol():
    global M,N
    cnt = 0
    for i in range(N):
        for j in range(M):
            if graph[i][j] == 1 and visited[i][j] == 0:
                dfs(i, j)
                cnt+=1
    return cnt

def dfs(row,col):
    visited[row][col] = 1
    for i in range(4):
        dx = col + d[i][0]
        dy = row + d[i][1]
        if 0 <= dx < M and 0 <= dy < N and visited[dy][dx] == 0 and graph[dy][dx] == 1:
            dfs(dy,dx)



T = int(input())

for _ in range(T):
    M, N, K = map(int, input().split())
    graph = [[0 for _ in range(M)] for _ in range(N)]
    visited = [[0 for _ in range(M)] for _ in range(N)]
    for _ in range(K):
        x,y = map(int, input().split())
        graph[y][x] = 1
    print(sol())

File: Graph/1012.cpp
#include <queue>
#include <iostream>
#include <vector>
#include <algorithm>
#include <queue>

using namespace std;

void bfs(int x, int y){
    queue<pair<int,int>> deque;
    pair<int,int> =

    while(deque.empty()){
        pair<int,int> p;
        p = deque.front();


    }
}

void sol(){
    int m,n,k;
    int cnt = 0;

    cin >> m >> n >> k;

    int graph[50][50];
    int visited[50][50];
    memset(graph, 0, sizeof(graph));
    memset(visited, 0, sizeof(visited));


    for (int i =0 ; i < k ; i++){
        int x,y;
        cin >> x >> y;
        graph[y][x] = 1;
    }

    for (int i =0 ; i < n ; i++){
        for (int j= 0 ; i< m ; j++){
            if(graph[i][j] == 1 && visited[i][j] == 0)
                bfs(j,i);
                cnt+=1;
        }
    }

}
int main(void){
    int T;
    cin >> T;

    for (int i =0; i< T ; i++){
        sol();
    }
}